src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -2742,6 +2742,36 @@ private function registerMailerConfiguration(array $config, ContainerBuilder $co
             $container->removeDefinition('mailer.messenger_transport_listener');
         }
 
+        if ($config['dkim_signer']['enabled']) {
+            $dkimSigner = $container->getDefinition('mailer.dkim_signer');
+            $dkimSigner->setArgument(0, $config['dkim_signer']['key'] ?? null);
+            $dkimSigner->setArgument(1, $config['dkim_signer']['domain'] ?? null);
+            $dkimSigner->setArgument(2, $config['dkim_signer']['select'] ?? null);
+            $dkimSigner->setArgument(3, $config['dkim_signer']['options'] ?? []);
+            $dkimSigner->setArgument(4, $config['dkim_signer']['passphrase'] ?? null);
```

Comment:
instead of repeating defaults here, define them properly in the Configuration class and use `->addDefaultsIfNotSet` on the arrayNode. Having the default values defined in the Configuration class provides a better DX as it means that tools like `dump:config-reference` can know them.